framework/db/ActiveQueryTrait.php

```diff
@@ -162,7 +162,7 @@ public function findWith($with, &$models)
     {
         $primaryModel = reset($models);
         if (!$primaryModel instanceof ActiveRecordInterface) {
-            $primaryModel = new $this->modelClass();
+            $primaryModel = \Yii::createObject($this->modelClass);
```

Comment:
This should be solved using `StaticInstanceInterface` instead. In case `$this->modelClass` implements  `yii\base\StaticInstanceInterface` (can be checked with http://php.net/manual/en/function.is-subclass-of.php ), call static instance method. If it does not implement the interface, use the current code with `new`.